Added HeaderAndPacketMarshalSize

sirzooro opened the pull request:
`PacketMarshalSize` was added in PR https://github.com/pion/rtp/pull/307 which was merged two days ago. During implementation of another part of that fix for `pion/srtp` I have realized that it would be handy to return header size too to avoid calculating it again.

JoeTurki reviewed (approved):
I feel it would've been more appropriate to have this as a method in the header struct, to me this feels like a hack having this as a global function, PacketMarshalSize kinda made sense but i feel this fits more in headers file as `func (h *Header) HeaderAndPacketMarshalSize()`, Not sure.

Thank you for doing this project.

sirzooro commented:
> I feel it would've been more appropriate to have this as a method in the header struct, to me this feels like a hack having this as a global function, PacketMarshalSize kinda made sense but i feel this fits more in headers file as `func (h *Header) HeaderAndPacketMarshalSize()`, Not sure.
> 
> Thank you for doing this project.

No, this will not work - Header does not contain payload. I have thought a bit about this and added new Packet2 struct, which contains pointer to Header instead of nesting it. I also added necessary methods to it, and marked global functions added previously as deprecated.

JoeTurki commented:
> No, this will not work - Header does not contain payload. I have thought a bit about this and added new Packet2 struct, which contains pointer to Header instead of nesting it. I also added necessary methods to it, and marked global functions added previously as deprecated.

I meant  `func (h *Header) HeaderAndPacketMarshalSize(payload)`, but your suggestion is even better, I think the only issue is that I don't think we should call it `Packet2` what about `FullPacket` or `PacketWithHeader` or any name you suggest, other than having versions in the code itself.

sirzooro commented on packet.go:
```diff
@@ -52,6 +52,16 @@ type Packet struct {
 }
 
+// Packet2 represents an RTP Packet. It is like Packet, but with header stored as a pointer.
+type Packet2 struct {
```
I have renamed it to `PacketWithHeaderPtr`.

JoeTurki replied:
Sorry for going back and forth on this, Do you think `Ptr` suffix is necessary? The way I see it that this a public API. and I'm not sure where this naming convention is from. I don't think we should have implementation details in a public API name.

JoeTurki replied:
This makes sense, I just don't want us to add global helpers just to fix issues with other packages, This should be just a RTP package.

Either ways I think RTP is due to a new major release soon, With WebRTC v5, Another thing that we should do is that we should separate all the codecs Packetizers and Depacketizer (like how it's now with av1 thanks to @jech suggestion).

sirzooro replied:
I am also not happy with adding new struct and global functions. Unfortunately changing Header to pointer is not backward compatible, so some workaround is needed. What do you think about using global functions only, and marking them as deprecated with note that they will be removed in WebRTC v5?

BTW, is there any TODO list and timeline for v5?

JoeTurki replied:
> What do you think about using global functions only, and marking them as deprecated with note that they will be removed in WebRTC v5?

This also works, do what you see fit.

> BTW, is there any TODO list and timeline for v5?

The current timeline is to push few minor releases to close backlog, and fix broken implementations, and some new features here and there, We aim to fix flex and HEVC by 5.2 next month, We talked about making a wiki for the stuff we need to change. I think we can aim for early 2026.

sirzooro replied:
Thanks.
